sources/ProfilesSessionPreferencesViewController.m

```diff
@@ -283,6 +285,11 @@ - (void)awakeFromNib {
             relatedView:nil
                    type:kPreferenceInfoTypeCheckbox];
 
+    [self defineControl:_progressBarHeight
+                    key:KEY_PROGRESS_BAR_HEIGHT
+            displayName:@"Progress bar height"
+                   type:kPreferenceInfoTypeIntegerTextField];
```

Comment:
One complication with this is that it will overlap terminal contents if it exceeds the size of the top margin (global setting kPreferenceKeyTopBottomMargins). You can assign to the `range` property of the `PreferenceInfo` that `defineControl:key:displayName:type:` returns to constrain it. But you'll want to observe changes to kPreferenceKeyTopBottomMargins and update the range appropriately. See iTermUserDefaultsObserver.